prime-router/src/main/kotlin/cli/MappingCommands.kt

```diff
@@ -0,0 +1,134 @@
+class MappingCommands : CliktCommand(
+    name = "data-mapping",
+    help = "CLI tool for data mapping tasks"
+) {
+    /**
+     * The directory to scan for unreferenced files.
+     * This option is required when the `--find-unreferenced` flag is provided.
+     */
+    private val inputDir by option("-d", "--directory", help = "Directory to scan for unreferenced files")
+        .file(true, canBeDir = true, mustBeReadable = true)
+
+    /**
+     * Output file to write the list of unreferenced files.
+     * If provided, the results will be written to this file instead of being printed to the console.
+     */
+    private val outputFile by option("-o", "--output-file", help = "Output file for unreferenced files")
+        .file()
+
+    /**
+     * Flag to trigger the action of finding unreferenced files in the specified directory.
+     * When this flag is set, the command will search for files that are not referenced by any other files in the directory.
+     */
+    private val findUnreferencedFiles by option(
+        "--find-unreferenced",
+        help = "Find unreferenced files in the directory"
+    )
+        .flag(default = false)
+
+    /**
+     * Main entry point for the command.
+     * If the `--find-unreferenced` flag is set, the command will find and print/write the unreferenced files.
+     * If no option is provided, a help message is displayed.
+     */
+    override fun run() {
+        // If findUnreferencedFiles option is set, call the corresponding function
+        if (findUnreferencedFiles) {
+            if (inputDir == null) {
+                throw CliktError("You must specify a directory with --directory to find unreferenced files.")
+            }
+            findUnreferencedFiles(inputDir!!)
+        } else {
+            echo("No command specified. Use --help to find available options.")
+        }
+    }
+
+    /**
+     * Main method to find unreferenced files in the given directory.
+     */
+    private fun findUnreferencedFiles(directory: File) {
+        if (!directory.isDirectory) {
+            throw CliktError("The specified path '${directory.absolutePath}' is not a directory.")
+        }
+
+        val allFiles = getAllFiles(directory)
+
+        // Extract relative file paths without extensions
+        val relativePathsWithoutExtension = allFiles.map {
+            it.relativeTo(directory).path.removeSuffix(".${it.extension}")
+        }
+
+        // Set of relative file paths that are referenced in other files
+        val referencedFiles = mutableSetOf<String>()
+
+        // Search for each relative path in other files
+        allFiles.forEach { file ->
+            referencedFiles.addAll(searchFileForReferences(file, relativePathsWithoutExtension))
+        }
+
+        // Find unreferenced files by checking which relative paths weren't found
+        val unreferencedFiles = allFiles.filter {
+            it.relativeTo(directory).path.removeSuffix(".${it.extension}") !in referencedFiles
```

Comment:
I think this logic should be updated to use something more precise like string matching. It appears that it is matching on part of a string ie if a file (`resource/Patient`) is contained within another (`resource/PatientMRG`) then it won't get logged as an unreferenced file. 

Backstory: I took a look at the files in /resources/ and it looked like Patient.yml wasn't being used anywhere. What I realized is that there is a line in patient that is getting matched on: `valueOf: resource/PatientMRG`. So there won't be false positives at least but this causes some files to be missed. 

After removing Patient.yml there were another 10 or so files that could be removed.